src/App.tsx

```diff
@@ -312,40 +326,54 @@ function App() {
-            explorerIsOpen={true}
-            showAttribution={true}
-          />
-          <GraphiQL
-            schema={state.schema}
-            query={state.currentOperation}
-            onEditQuery={onEdit}
-            storage={storage}
-            fetcher={() => Promise.resolve({ data: null })}
-          >
-            <GraphiQL.Toolbar>
-              <GraphiQL.Button
-                onClick={() => dispatch({ type: "prettifyOperation" })}
-                label="Prettify"
-                title="Prettify"
-              />
-              {command.type === "show" ? null : (
-                <GraphiQL.Button
-                  onClick={save}
-                  label="Save (Cmd/Ctrl + Enter)"
-                  title="Save (Cmd/Ctrl + Enter)"
-                />
-              )}
-              <GraphiQL.Button
-                onClick={cancel}
-                label={command.type === "show" ? "Close" : "Cancel (Esc)"}
-                title={command.type === "show" ? "Close" : "Cancel (Esc)"}
+          {state.show === "graphiql" ? (
+            <>
+              <GraphiQLExplorer
+                schema={state.schema}
+                query={state.currentOperation}
+                onEdit={onEditExplorer}
+                explorerIsOpen={true}
+                showAttribution={true}
               />
-            </GraphiQL.Toolbar>
-          </GraphiQL>
+              <GraphiQL
+                schema={state.schema}
+                query={state.currentOperation}
+                onEditQuery={onEdit}
+                storage={storage}
+                fetcher={() => Promise.resolve({ data: null })}
+              >
+                <GraphiQL.Toolbar>
+                  <GraphiQL.Button
+                    onClick={() => dispatch({ type: "prettifyOperation" })}
+                    label="Prettify"
+                    title="Prettify"
+                  />
+                  {command.type === "show" ? (
+                    <GraphiQL.Button
+                      onClick={() =>
+                        dispatch({ type: "setShowType", show: "voyager" })
+                      }
+                      label="Explore Graph"
+                      title="Explore Graph"
+                    />
+                  ) : (
+                    <GraphiQL.Button
+                      onClick={save}
+                      label="Save (Cmd/Ctrl + Enter)"
+                      title="Save (Cmd/Ctrl + Enter)"
+                    />
+                  )}
+                  <GraphiQL.Button
+                    onClick={cancel}
+                    label={command.type === "show" ? "Close" : "Cancel (Esc)"}
+                    title={command.type === "show" ? "Close" : "Cancel (Esc)"}
+                  />
+                </GraphiQL.Toolbar>
+              </GraphiQL>
+            </>
+          ) : state.show === "voyager" ? (
+            <Voyager introspection={{ data: introspectionFromSchema(state.schema) }} />
```

Comment:
Hmm, this is going to be problematic to do in render like this when used with large schemas. I think what's needed here is:

- Compute, memoize and save this to some state variable (`introspectionFromSchema(state.schema)`) when the user presses the button to show Voyager.
- While doing that, show some form of loading state. Just something indicating to the user that it's loading/converting the schema.

You can test this with the schema from OneGraph.com. If the experience with that is OK, most other schemas will work fine 😄